salesforcecli/plugin-source
# Sh/mdapi list cmd

### What does this PR do?
Adds the `force:mdapi:listmetadata` command to the source plugin with unit tests and NUTs.
Adds the `force:mdapi:describemetadata` command to the source plugin with unit tests and NUTs.

### What issues does this PR fix or reference?
@W-9603687@
@W-9603698@

QA Notes for mdapi:listmetadata:
1. Deploy some metadata, then run list commands
2. list -m
3. list -m --json
4. list -m --folder (requires deploying or creating inFolder metadata)
5. list -m -f (existing file and non-existing file)
6. list -m -a

Note: I changed the JSON output when `-f` flag is used because it should always return the result of the listmetadata call.  Returning the output file path is not particularly helpful, although maybe someone knows a good reason why it was done this way.
